### PR TITLE
ft: backbeat integration for zenko docker stack deployment

### PR DESCRIPTION
**Pull request by jonathan-gramain**
REVIEW ONLY

It's relying on some base images in zenko docker hub, and some backbeat and arsenal PRs not yet merged, so review only for now.

### CODE REVIEW

File: images/kafka/server.properties.backbeat [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+num.io.threads=8
+
+# The send buffer (SO_SNDBUF) used by the socket server
+socket.send.buffer.bytes=102400
+
+# The receive buffer (SO_RCVBUF) used by the socket server
+socket.receive.buffer.bytes=102400
+
+# The maximum size of a request that the socket server will accept (protection against OOM)
+socket.request.max.bytes=104857600
```
Comment (LaurenSpiegel) [inline images/kafka/server.properties.backbeat]: how were these numbers determined?

File: swarm-production/docker-stack.yml [hunk 2]
```diff
@@ -76,27 +79,73 @@ services:
...
+      mode: replicated
+      placement:
+        constraints:
+          - node.labels.io.zenko.type == storage
+
+  backbeat:
+    image: zenko/backbeat
+    networks:
+      - backend
+    command: npm start
```
Comment (LaurenSpiegel) [inline swarm-production/docker-stack.yml]: is this supposed to start a task? how do we determine whether we are starting the replication task or the clueso task?

File: images/kafka/server.properties.backbeat (hunk 1 above)
Comment (jonathan-gramain) [reply images/kafka/server.properties.backbeat]: Those values are from the default kafka config file, so I assume they are sensible default values (actually I started from Vianney's branch without looking closely at this file except the critical params but that does not waive the need for an explanation :smile: )

File: swarm-production/docker-stack.yml (hunk 2 above)
Comment (jonathan-gramain) [reply swarm-production/docker-stack.yml]: I added the `start` rule in https://github.com/scality/backbeat/pull/88, which starts both the queue populator and the replication processor. We would need to selectively start the services that we'll need at processing stage, say replication/clueso/both, though I don't know what's the best way to deal with this at this point. Any idea?

Comment (jonathan-gramain) [reply swarm-production/docker-stack.yml]: Note that on the populator side I assume there will be one populator for all services, with pluggable filters for each service (again this has to be properly designed)

Comment (LaurenSpiegel) [reply swarm-production/docker-stack.yml]: I like the idea of pluggable filters. Clueso has other tasks beyond the queue (creating the METADATA bucket for instance). Maybe there could be environment variables here for clueso or replication and if those are set, in backbeat we then start whatever processes beyond the populator are necessary?

**Comment by ssalaues** [comment]
Since backbeat has been merged into both stacks, I will be closing this issue. Please feel free to reopen if needed.

**Pull request closed without merging**